model/scratch/dffml_model_scratch/sagaopt.py

```diff
@@ -0,0 +1,178 @@
+import numpy as np
+import warnings
+from sklearn.linear_model.sag_fast import sag32, sag64
+from sklearn.utils.seq_dataset import ArrayDataset32, ArrayDataset64

```

Comment:
If we're using scikit in then we should put these files under `model/scikit`

Reply:
These are the optimizers and dataset framers only

Reply:
done.